pkg/engine/jmespath/functions.go

```diff
@@ -0,0 +1,351 @@
+package jmespath
+
+import (
+	"errors"
+	"regexp"
+	"strconv"
+	"strings"
+
+	gojmespath "github.com/jmespath/go-jmespath"
+)
+
+var (
+	JpObject      = gojmespath.JpObject
+	JpString      = gojmespath.JpString
+	JpNumber      = gojmespath.JpNumber
+	JpArray       = gojmespath.JpArray
+	JpArrayString = gojmespath.JpArrayString
+)
+
+type (
+	JpType  = gojmespath.JpType
+	ArgSpec = gojmespath.ArgSpec
+)
+
+func getFunctions() []*gojmespath.FunctionEntry {
+	return []*gojmespath.FunctionEntry{
+		{
+			Name: "compare",
+			Arguments: []ArgSpec{
+				{Types: []JpType{JpString}},
+				{Types: []JpType{JpString}},
+			},
+			Handler: jpfCompare,
+		},
+		{
+			Name: "contains",
+			Arguments: []ArgSpec{
+				{Types: []JpType{JpString}},
+				{Types: []JpType{JpString}},
+			},
+			Handler: jpfContains,
+		},
+		{
+			Name: "equal_fold",
+			Arguments: []ArgSpec{
+				{Types: []JpType{JpString}},
+				{Types: []JpType{JpString}},
+			},
+			Handler: jpfEqualFold,
+		},
+		{
+			Name: "replace",
+			Arguments: []ArgSpec{
+				{Types: []JpType{JpString}},
+				{Types: []JpType{JpString}},
+				{Types: []JpType{JpString}},
+				{Types: []JpType{JpNumber}},
+			},
+			Handler: jpfReplace,
+		},
+		{
+			Name: "replace_all",
+			Arguments: []ArgSpec{
+				{Types: []JpType{JpString}},
+				{Types: []JpType{JpString}},
+				{Types: []JpType{JpString}},
+			},
+			Handler: jpfReplaceAll,
+		},
+		{
+			Name: "to_upper",
+			Arguments: []ArgSpec{
+				{Types: []JpType{JpString}},
+			},
+			Handler: jpfToUpper,
+		},
+		{
+			Name: "to_lower",
+			Arguments: []ArgSpec{
+				{Types: []JpType{JpString}},
+			},
+			Handler: jpfToLower,
+		},
+		{
+			Name: "trim",
+			Arguments: []ArgSpec{
+				{Types: []JpType{JpString}},
+				{Types: []JpType{JpString}},
+			},
+			Handler: jpfTrim,
+		},
+		{
+			Name: "split",
+			Arguments: []ArgSpec{
+				{Types: []JpType{JpString}},
+				{Types: []JpType{JpString}},
+			},
+			Handler: jpfSplit,
+		},
+		{
+			Name: "equals",
+			Arguments: []ArgSpec{
+				{Types: []JpType{JpString}},
+				{Types: []JpType{JpString}},
+			},
+			Handler: jpEquals,
+		},
+		{
+			Name: "regexReplaceAll",
+			Arguments: []ArgSpec{
+				{Types: []JpType{JpString}},
+				{Types: []JpType{JpString, JpNumber}},
+				{Types: []JpType{JpString, JpNumber}},
+			},
+			Handler: jpRegexReplaceAll,
+		},
+		{
+			Name: "regexReplaceAllLiteral",
+			Arguments: []ArgSpec{
+				{Types: []JpType{JpString}},
+				{Types: []JpType{JpString, JpNumber}},
+				{Types: []JpType{JpString, JpNumber}},
+			},
+			Handler: jpRegexReplaceAllLiteral,
+		},
+		{
+			Name: "regexMatch",
+			Arguments: []ArgSpec{
+				{Types: []JpType{JpString}},
+				{Types: []JpType{JpString, JpNumber}},
+			},
+			Handler: jpRegexMatch,
+		},
+	}
+
+}
+
+func jpfCompare(arguments []interface{}) (interface{}, error) {
+	a, ok := arguments[0].(string)
+	if !ok {
+		return nil, errors.New("Compare: first argument is expected of string type")
```

Comment:
For all errors lets add `JMESPath function <name>:` as a prefix.  The name should exactly match the function name (including case).

Reply:
Done